patches/tModLoader/Terraria/ModLoader/Core/AssemblyManager.cs

```diff
@@ -183,6 +183,21 @@ private static Assembly hook_ValidateAssemblyNameWithSimpleName(orig_ValidateAss
 			GC.Collect();
 			GC.WaitForPendingFinalizers();
 		}
+
+		var type = typeof(MonoMod.Utils.ReflectionHelper);
+		FieldInfo[] caches = [
+			type.GetField("AssemblyCache", BindingFlags.NonPublic | BindingFlags.Static),
+			type.GetField("AssembliesCache", BindingFlags.NonPublic | BindingFlags.Static),
+			type.GetField("ResolveReflectionCache", BindingFlags.NonPublic | BindingFlags.Static),
+		];
+		foreach(var cache in caches) {
+			if(cache == null) {
+				continue;
+			}
+			if (cache.GetValue(null) is System.Collections.IDictionary dict) {
```

Comment:
We don't want to code defensively here, because `MonoMod` is a dependency that we tightly depend on and control, and we don't have a test suite that could catch regressions from updates. 

In cases where external code is involved, defensive programming is warranted, but we would still want to log a warning rather than silently failing.

Suggest:
```cs
((IDictionary)cache.GetValue(null)).Clear();
```